{
  "meta":
  {
    "name": "mappingWithError",
	"description": "Execute a workflow containing a mapping (loaded from app) that throws an exception and open audit",
	"created": "11.12.2019",
	"updated": "30.03.2020"
  },
  "operations":
  [
    {
	  "operation": "set",
	  "variable": "!customFieldValues!",
	  "value": ["test", "", "", ""]
	},
    {
      "operation": "invoke",
      "functionName": "runWorkflow",
      "mapInput": 
	  {
	    "!workflowFQN!": "xmcp.h5x.regression.MappingWorkflowWithError",
		"!input!": [],
		"!customStringContainer!": "!customFieldValues!"
	  },
      "mapOutput": 
	  { 
	    "!orderId!": "!orderId!",
		"!applicationName!": "!applicationName!",
		"!applicationVersion!": "!applicationVersion!"
      }
    },
	{
	  "operation": "invoke",
	  "functionName": "assertGeneralFields",
	  "mapInput":
	  {
	    "!orderId!": "!orderId!",
		"!dataflowLength!": 1,
		"!retryLength!": 0,
		"!errorsLength!": 1,
		"!rollbackLength!": 0,
		"!customFieldValues!": "!customFieldValues!",
		"!applicationName!": "!applicationName!",
		"!applicationVersion!": "!applicationVersion!"
	  }
	},
	{
      "operation": "read",
	  "targetVariable": "!exception!",
	  "pathInResponse": "info/$list[id=step9]/error/exception",
	  "unquoteResult": true
	},
	{
	  "operation": "assert",
	  "constraint": "equals",
	  "variable": "!exception!",
	  "value": "java.lang.Exception"
	},
	{
      "operation": "read",
	  "targetVariable": "!status!",
	  "pathInResponse": "info/$list[id=step9]/status",
	  "unquoteResult": true
	},
	{
	  "operation": "assert",
	  "constraint": "equals",
	  "variable": "!status!",
	  "value": "Failed"
	}
  ]
}